client/src/shared/state/Inventory.js

```diff
@@ -104,16 +110,76 @@ class Inventory {
         if (!ItemTypes[itemConfig.typeCode].hasUseEffect) return;
 
         this.hotbar.push(itemConfig);
-
+        this.saveItemId[itemConfig.id] = itemConfig.slotIndex;
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
     }
 
     removeFromHotbar(itemConfig) {
+        delete this.saveItemId[itemConfig.id];
         this.hotbar = this.hotbar.filter((eachItem) => eachItem !== itemConfig);
-
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
+    }
+
+    saveHotbar() {
+        window.localStorage.setItem(this.loadHotBarRequest, JSON.stringify(this.saveItemId));
+    }
+
+    defaultHotBar() {
+        this.items.forEach((itemConfig) => {
+            this.addToHotbar(itemConfig);
+        });
+    }
+
+    loadFromLocalStorage(playerID) {
+        this.saveSession = true;
+        this.userName = playerID;
+        const gameName = "dungeonz";
+        const requestHotbar = "hotBar";
+        this.loadHotBarRequest = gameName + this.userName + requestHotbar;
```

Comment:
What is the purpose of this `gameName` and `requestHotbar` namespacing?

Reply:
For local storage, if the user was to inspect their localStorage and see what is saved, they know that it is coming from the game dungeons, and the requestHotbar was just to specify that it is saving the Hotbar.

Reply:
I think it is explicit enough if they are on the game that it is from the game "dungeonz", as local storage is not shared across domains, so there won't be other games in local storage to confuse it with.

Reply:
So what would be a good name for the key, just userName + Hotbar, or just userName?

Reply:
userName + Hotbar should be fine.
or, accountID + Hotbar.
I think "userName" is misleading here, as it isn't using the account username you log in with.
The data being loaded for that property is just the DB document ID for the account, which is `accountID`.

It reads a bit confusing across this PR to see `accountID`, `playerID`, and `userName`, even though they are all using the same value.

Reply:
Ok from now on follow just accountID as the main unique identifier.

Reply:
Also, there is a convention already for the local storage keys, being `some_key_name`. See client/src/game/Boot.js for current uses of localStorage and how the keys look.

Reply:
Also, the linter expects template literals to be used for this kind of string.
so
```js
this.loadHotBarRequest = this.accountID + "_hotbar";
```
would be written as
```js
this.loadHotBarRequest = `${this.userName}_hotbar;
```
Depending on your IDE, it should warn you about formatting mistakes like this. For VSCode you may need the ESLint plugin though.